index.js

```diff
@@ -87,14 +89,44 @@ module.exports = function loader(content) {
       }
     });
 
+    if (outputPlaceholder) {
+      q.defer(function generatePlaceholder(queueCallback) {
+        img
+            .clone()
+            .resize(placeholderSize, jimp.AUTO)
+            .quality(quality)
+            .background(background)
+            .getBuffer(mime, function resizeCallback(queueErr, buf) {
+              if (err) {
+                return queueCallback(queueErr);
+              }
+
+              const placeholder = buf.toString('base64');
+              return queueCallback(null, JSON.stringify('data:' + (mime ? mime + ';' : '') + 'base64,' + placeholder));
+            });
+      });
+    }
+
     return q.awaitAll((queueErr, files) => {
+      'use strict'; // eslint-disable-line
+      let placeholder;
+      if (outputPlaceholder) {
+        placeholder = files.pop();
+      }
+
       const srcset = files.map(f => f.src).join('+","+');
 
       const images = files.map(f => '{path:' + f.path + ',width:' + f.width + ',height:' + f.height + '}').join(',');
 
       const firstImagePath = files[0].path;
 
-      loaderCallback(null, 'module.exports = {srcSet:' + srcset + ',images:[' + images + '],src:' + firstImagePath + ',toString:function(){return ' + firstImagePath + '}};');
+      loaderCallback(null, 'module.exports = {' +
```

Comment:
Yes, still too jet lagged. I've added the _original_ image dimensions to the output.